tests/unit_tests/test_errors.py

```diff
@@ -0,0 +1,179 @@
+##############################################################################
+# (c) Crown copyright Met Office. All rights reserved.
+# For further details please refer to the file COPYRIGHT
+# which you should have received as part of this distribution
+##############################################################################
+"""
+Unit tests for custom fab exceptions.
+"""
+
+import pytest
+from unittest.mock import Mock, PropertyMock
+
+from fab.errors import (
+    FabError,
+    FabToolError,
+    FabToolMismatch,
+    FabToolInvalidVersion,
+    FabToolPsycloneAPI,
+    FabToolNotAvailable,
+    FabToolInvalidSetting,
+    FabCommandError,
+    FabCommandNotFound,
+    FabMultiCommandError,
+    FabSourceNoFilesError,
+    FabSourceMergeError,
+    FabSourceFetchError,
+    FabUnknownLibraryError,
+)
+
+
+class TestErrors:
+    """Basic tests for the FabError class hierarchy."""
+
+    def test_base(self):
+        """Test the base Fab error class."""
+
+        err = FabError("test message")
+        assert str(err) == "test message"
+
+    def test_unknown_library(self):
+        """Test unknown library errors."""
+
+        err = FabUnknownLibraryError("mylib")
+        assert str(err) == "unknown library mylib"
+
+
+class TestToolErrors:
+    """Test the FabToolError hierarchy."""
+
+    def test_tool_string(self):
+        """Test the base FabToolError class."""
+
+        err = FabToolError("cc", "compiler message")
+        assert str(err) == "[cc] compiler message"
+
+        # Mock defines an internal name property, so special handling
+        # is required to reset the value to support testing
+        tool = Mock(_name="tool cc")
+        del tool.name
+        err = FabToolError(tool, "compiler message")
+        assert str(err) == "[tool cc] compiler message"
```

Comment:
This seems alarming. Is this related to Joerg's comment about peeking at private state?